corehq/apps/users/views/mobile/users.py

```diff
@@ -678,6 +676,77 @@ def post(self, request, *args, **kwargs):
         return self.get(request, *args, **kwargs)
 
 
+# This is almost entirely a duplicate of CreateCommCareUserView. That view will
+# be going away soon, so I didn't bother to abstract out the commonalities.
+class CreateCommCareUserModal(JsonRequestResponseMixin, DomainViewMixin, View):
```

Comment:
When the user hits the button, it pops up a modal which immediately pings this view - this view sends down a rendered html form and does the validation all inside the modal.